Add support for Fuel's Query::cached()

Meroje opened the pull request:
This adds a new method query_cache that will pass through to cached(), same api.  
Passing true will fetch page's duration from the app config as default lifetime.

Commit:
Add query_cache that passes through to fuel's Query::cached()

Passing only true will use default cache duration (same as pages)

Foine commented:
I think that add a config that can be activated or not can be a good idea to not force user to have this cache function.

Also, it seems that indentation is broken (well, it seems like it when i see files changes on github)

Commit:
Fix indenting, change default query lifetime to new config key

Meroje commented:
I changed to a new config 'novius-os.cache_duration_query', not setting it but still requesting a query_cache(true) will result in a 0 seconds cache.

Foine commented:
The config can't be `novius-os.cache_duration_query`. You are in the fuelphp ORM here. Novius OS "don't exist" here. I suggest you to call the ORM config.

vrcAlbert commented:
@Foine is right, otherwise it seems good to me :+1:

Commit:
Pass through everything, except false that disables the call to Query::cached

Lets the cache driver handle default expiration and invalid $lifetime values (not numeric)

Foine commented:
:+1:
